src/alchemlyb/parsing/namd.py

```diff
@@ -25,72 +108,162 @@ def extract_u_nk(fep_file, T):
+                # within the same file, then complain. This can happen if truncated fepout files
+                # are presented in the wrong order.
+                if l[0] == '#NEW':
+                    lambda1_at_start, lambda2_at_start = float(l[6]), float(l[8])
+                    lambda_idws_at_start = float(l[10]) if 'LAMBDA_IDWS' in l else None
+                    has_idws = True if lambda_idws_at_start is not None else False
+
+                # this line marks end of window; dump data into dataframe
+                if l[0] == '#Free':
+                    # extract lambda values for finished window
+                    # lambda1 = sampling lambda (row), lambda2 = comparison lambda (col)
+                    lambda1 = float(l[7])
+                    lambda2 = float(l[8])
+
+                    # If the lambdas are not what we thought they would be, raise an exception to ensure the calculation
+                    # fails. This can happen if fepouts where one window spans multiple fepouts are processed out of order
+                    # NB: There is no way to tell if lambda_idws changed because it isn't in the '#Free' line that ends a window
+                    if lambda1_at_start is not None \
+                        and (lambda1, lambda2) != (lambda1_at_start, lambda2_at_start):
+                        logger.error(f"Lambdas changed unexpectedly while processing {fep_file}")
+                        logger.error(f"l1, l2: {lambda1_at_start}, {lambda2_at_start} changed to {lambda1}, {lambda2}")
+                        logger.error(line)
+                        raise ValueError("Inconsistent lambda values within the same window")
+
+                    # As we are at the end of a window, convert last window's work and times values to np arrays
+                    # (with energy unit kT since they were kcal/mol in the fepouts)
+                    win_de_arr = beta * np.asarray(win_de) # dE values
+                    win_ts_arr = np.asarray(win_ts) # timesteps
+
+                    # Infer lambda_idws_at_start if it wasn't explictly included in this fepout.
+                    # If lambdas are in ascending order, choose the one before it
+                    # This can happen if user ran minimize before starting dynamics and didn't do firsttimestep 0,
+                    # because NAMD only emits the '#NEW' line on timestep 0 for some reason
+                    if has_idws and lambda_idws_at_start is None:
+                        l1_idx, l2_idx = all_lambdas.index(lambda1), all_lambdas.index(lambda2)
+                        if l1_idx > 0 and l1_idx < l2_idx: # Ascending lambdas
+                            lambda_idws_at_start = all_lambdas[l1_idx - 1]
+                        elif l2_idx < (len(all_lambdas) - 1) and l2_idx < l1_idx: # Descending lambdas
+                            lambda_idws_at_start = all_lambdas[l2_idx + 1]
```

Comment:
When should this conditional be triggered? Is this a common or a rare thing?

At the moment this case is not covered by tests.

Reply:
This would occur when we parse a FEP calculation result where lambda progresses 1 -> 0 over the set of windows, rather than (to me) the usual case of 0 -> 1. This bit of code infers what the lambda_idws likely is, in the event no lambda_idws is present in this .fepout file - which can happen when a NAMD run is interrupted.

Hm, I guess we'd want to test a 1 -> 0 FEP calculation?

Reply:
If you as the expert say that this could happen in the wild then we want a test — yes, please.

Reply:
OK. In this case the easiest thing for me is to add another test set in alchemtest. The format of NAMD .fepout files that use IDWS makes it difficult to modify the existing test set on the fly as is done to trigger the other failure cases: lambda values are in both header and footer, but lambda_idws is not in footer. So swapping them around across fragments of .fepout files that would arise from an interrupted and restarted simulation would make the test code more complex than the code it's meant to test.

Reply:
If you add it to alchemtest and ping me I'll review it as quickly as possible and try to get it in ASAP.